### ISSUE TITLE
Basic question about your docker run

### ISSUE THREAD

**Issue by Node815** [TheManchineel/vaultwarden-api#1]
Is this the $argo2 token or the one you enter in when you go the /admin page?

-e VAULTWARDEN_TOKEN=your_vaultwarden_token \

I tried both the regular login I use when entering the /admin page and then the $argo2 one from my compose file which Vaultwarden uses and it spams the login enough to get locked out of the admin page unless I reboot the server I'm running it on to clear the failures.

**Comment by TheManchineel** [comment]
It should be the password you use to access the admin page itself, not its hash.

 * Have you checked that you can otherwise login successfully with said password?
 * Does the password contain any special characters that you're not escaping from within your shell?
 * Could you please post an anonymized, minimal example of your current compose file for your stack? I'd love to test this on my end

This is basically a toy project that I created for my own personal use so unfortunately it's quite rough right now, but I'll gladly give this a look once you check the above :)

**Comment by Node815** [comment]
I was able to login with my regular password in the admin page. 
Regular password does use the Exclamation point within it. 

```version: '3'
services:
  vaultwarden:
    container_name: vaultwarden
    image: vaultwarden/server:latest
    restart: unless-stopped
    volumes:
      - /mnt/docker/appdata/vaultwarden/vw-data/:/data/
    ports:
      - 80:80
    environment:
      - DOMAIN=https://my.domainvaultexample.com
      - LOGIN_RATELIMIT_MAX_BURST=10
      - LOGIN_RATELIMIT_SECONDS=60
      - ADMIN_RATELIMIT_MAX_BURST=10
      - ADMIN_RATELIMIT_SECONDS=60
      - ADMIN_TOKEN=${VAULTWARDEN_ADMIN_TOKEN}
      - SENDS_ALLOWED=true
      - EMERGENCY_ACCESS_ALLOWED=true
      - WEB_VAULT_ENABLED=true
      - SIGNUPS_ALLOWED=false
      - SIGNUPS_VERIFY=false
      - SIGNUPS_VERIFY_RESEND_TIME=3600
      - SIGNUPS_VERIFY_RESEND_LIMIT=5
      - SIGNUPS_DOMAINS_WHITELIST=
      - SMTP_HOST=
      - SMTP_FROM=
      - SMTP_FROM_NAME=Vaultwarden
      - SMTP_SECURITY=starttls
      version: '3'
```

I stripped out my email stuff.  The admin token is stored in an .env file with:
```VAULTWARDEN_ADMIN_TOKEN='$argon2id$THERESTOFMYTOKEN'```

**Comment by TheManchineel** [comment]
So there seems to be nothing wrong with your current configuration. For the sake of troubleshooting, could you try lowering the ADMIN_RATELIMIT_SECONDS to 5 or lower? (it is a cap on maximum consecutive logins, which affects successful as well as failed logins to the admin interface).

You might also wanna check (if you're using Cloudflare) that you aren't using Scrape Shield on your Vaultwarden domain (particularly the "Server-side Excludes" and "Email Address Obfuscation") features, I remember that one causing me trouble earlier. You can either disable them altogether temporarily, or create a more permanent Configuration Rule to disable them for your specific Vaultwarden subdomain and/or for requests originating from your own IP address/range. In general I'd try turning off most of Cloudflare protections for connections that are from your own network.

Finally, if you have Python with the `requests` package installed (`pip3 install requests`), I'd be curious about the output of:

```py
import requests

VAULTWARDEN_ADMIN_URL="https://vault.example.com/admin" # replace your domain here
VAULTWARDEN_PASSWORD="1234abcd!" # replace your key here

response = requests.post(VAULTWARDEN_ADMIN_URL, data={"token": VAULTWARDEN_PASSWORD}))
print(response.status_code)
print(response.cookies["VW_ADMIN"])
```

You should get:

```
200
[ followed by a SECRET temporary token you should NOT share with anyone else ]
```

If you get a `KeyError` or anything similar, it means that there is a problem when programmatically authenticating to your VW instance. Cloudflare might be the culprit.

**Comment by Node815** [comment]
I found a way to authenticate at least by going to the Key settings under your user in Vaultwarden, it will show something like this:
![image](https://github.com/TheManchineel/vaultwarden-api/assets/20736534/d77a068c-df21-4eec-bd7d-7e4cab8ca7d7)

I put the client secret in the API environment variable.  Then the my plain text password in the VAULTWARDEN_TOKEN environment.  No errors or lockouts.  But I know they don't work when I tested both client secret and client ID from the info given from Vaultwarden's keys.   I figured I toss this your way,  I'm done for now, I've been toying with moving from Vaultwarden to get full OIDC in my stack and that's the only thing holding me back.
